### PR TITLE
Add to Cart Back End 

### PR DESCRIPTION
**Pull request by jeff-jones2020**
image

image

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -49,6 +49,86 @@ app.get('/api/products/:id', (req, res, next) => {
...
+      "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using ("productId")
+    where "c"."cartItemId" = $1
+  `;
+
+  db.query(priceSql, [productId])
+    .then(result => {
+      if (result.rowCount === 0) {
+        return next(new ClientError(`Product ID '${productId}' does not exist`, 404));
```
Comment (thebearingedge) [inline server/index.js]: When you're in a promise chain, returning isn't enough to stop the chain. You need to `throw new ClientError(...)` or return `Promise.reject(new ClientError(...))`

File: server/index.js [hunk 2]
```diff
@@ -64,7 +144,7 @@ app.use((err, req, res, next) => {
   }
 });
 
-app.listen(process.env.PORT, () => {
+app.listen(3000, () => {
```
Comment (thebearingedge) [inline server/index.js]: Please do not hard-code the port. It should be defined in your `.env` file

File: server/index.js [hunk 3]
```diff
@@ -64,7 +144,7 @@ app.use((err, req, res, next) => {
...
   // eslint-disable-next-line no-console
-  console.log('Listening on port', process.env.PORT);
+  console.log('Listening on port', 3000);
```
Comment (thebearingedge) [inline server/index.js]: Please do not hard-code the port. It should be defined in your .env file

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 2 above)
Comment (jeff-jones2020) [reply server/index.js]: Fixed

File: server/index.js (hunk 1 above)
Comment (jeff-jones2020) [reply server/index.js]: Changed

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**